FWCore/Framework/interface/ComponentMaker.h

```diff
@@ -92,12 +96,28 @@ namespace edm {
     std::shared_ptr<typename ComponentMaker<T, TComponent>::base_type> ComponentMaker<T, TComponent>::addTo(
         EventSetupsController& esController,
         EventSetupProvider& iProvider,
-        ParameterSet const& iConfiguration,
+        ParameterSet& iConfiguration,
         bool replaceExisting) const {
       // This adds components to the EventSetupProvider for the process. It might
       // make a new component then add it or reuse a component from an earlier
       // SubProcess or the top level process and add that.
 
+      {
+        auto modtype = iConfiguration.getParameter<std::string>("@module_type");
+        auto moduleLabel = iConfiguration.getParameter<std::string>("@module_label");
+        ConfigurationDescriptions descriptions(T::baseType(), modtype);
+        fillDetails::fillIfExists<TComponent>(descriptions);
```

Comment:
can any of these throw, even just for a bad alloc?

Reply:
I'd expect at least `bad_alloc` to be possible. The code structure is pretty much just moved from `edm::eventsetup::validateEventSetupParameters()`. Are you suggesting to enclose all this code in the try-converException block below?

Reply:
yes

Reply:
Done